sailfishusermanager.cpp

```diff
@@ -304,6 +317,10 @@ void SailfishUserManager::removeUser(uint uid)
         return;
     }
 
+    struct passwd *pwd = getpwuid(uid);
+    if (pwd)
+        removeUserFiles(pwd->pw_name);
```

Comment:
uid -> name via getpwuid(), then that returned name -> uid via getpwnam() = potentially disastrous pitfall given the context of recursive directory wiping

"... return value may point to a static area, and may be overwritten by subsequent calls to getpwent(3), getpwnam(), or getpwuid()."